File: src/main/java/org/mitallast/queue/crdt/replication/Replicator.java
package org.mitallast.queue.crdt.replication;

import org.mitallast.queue.common.codec.Message;
import org.mitallast.queue.common.component.LifecycleComponent;
import org.mitallast.queue.crdt.protocol.AppendRejected;
import org.mitallast.queue.crdt.protocol.AppendSuccessful;

public interface Replicator extends LifecycleComponent {

    void append(long id, Message event);

    void successful(AppendSuccessful message);

    void rejected(AppendRejected message);

    void open();

    void closeAndSync();
}

File: src/main/java/org/mitallast/queue/crdt/replication/Replicator.kt
package org.mitallast.queue.crdt.replication

import org.mitallast.queue.common.codec.Message
import org.mitallast.queue.common.component.LifecycleComponent
import org.mitallast.queue.crdt.protocol.AppendRejected
import org.mitallast.queue.crdt.protocol.AppendSuccessful

interface Replicator : LifecycleComponent {

    fun append(id: Long, event: Message)

    fun successful(message: AppendSuccessful)

    fun rejected(message: AppendRejected)

    fun open()

    fun closeAndSync()
}
